includes/class-wc-structured-data.php

```diff
@@ -275,38 +274,46 @@ public function generate_product_data( $product = null ) {
 				'reviewCount' => $product->get_review_count(),
 			);
 
-			// Markup most recent rating/review.
+			// Markup 5 most recent rating/review.
 			$comments = get_comments(
 				array(
-					'number'      => 1,
+					'number'      => 5,
```

Comment:
Nice touch to limit the number of reviews to avoid performance issues.

Considering that we are already loading the reviews anyway in the product single page, I wonder if we call `get_comments()` using exactly the same arguments that were used when loading the reviews (or maybe even use a method that calls `get_comments()` internally). This way, when calling it the second time, WP would return a cached response instead of hitting the database twice to get more or less the same data. What do you think of this approach? No worries if it doesn't seem easy to implement or worth it pursuing.

Reply:
I feel the same, and I did look a little at `get_comments()` as well, but we need to skip non-reviews, something that we allow in the front-end, in my changes I also changed the query, so we avoid call any non-review, or reviews with no stars. Because of that I don't think that we could make WP return cached response, or at least should already return for the comment objects in `get_comment_x()` functions.